### PR TITLE
InfluxDB: Add http configuration when selecting InfluxDB v2 flavor

### LINKED ISSUE #29303: Influx: There is not auth configuration when Flux language is selected
**What happened**:
Can't connect to influxdb 2.0 when using flux datasource

**What you expected to happen**:
Somewhere to set https settings or certs

**How to reproduce it (as minimally and precisely as possible)**:
Add Flux datasource when influx is setup using TLS

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.3.2
- Data source type & version: Influx Version 2.0.0 (37cc047)
- OS Grafana is installed on: ubuntu
- User OS & Browser: chrome

### PR DESCRIPTION
**Pull request by aocenas**
Fixes: https://github.com/grafana/grafana/issues/29303

Adds the HTTP default config section. Also adds basic InfluxDB2 dev env block for testing.

### CODE REVIEW

**Review by ivanahuckova — APPROVED** [review]
Question - not necessarily to this PR, but when I want to add Flux data source, the basic auth toggle is already switched on, which I think is not necessary, as Flux data source works even if I turn it off. 

Awesome that you have added flux docker block. I wasn't able to test it, but that is probably just my problem with generally running docker blocks (I have to figure that out). 

I would still wait for @ryantxu to do a review, but from me - LGTM.

**Comment by aocenas** [comment]
> the basic auth toggle is already switched on

Honestly not sure about that one it was already there in the code so did not change that. @ryantxu do you know if there was a purpose for it?

File: devenv/docker/blocks/influxdb2/docker-compose.yaml
Comment (papagian) [inline devenv/docker/blocks/influxdb2/docker-compose.yaml:17]:
```suggestion
    entrypoint: influx setup --bucket mybucket -t mytoken -o myorg --username=grafana --password=grafana12345 --host=http://influxdb:8086 -f
```
otherwise I get an error like:
`Error: failed to setup instance: password is too short`

Comment (aocenas) [reply devenv/docker/blocks/influxdb2/docker-compose.yaml]: Thanks, seems like I forgot to push the update

File: public/app/plugins/datasource/influxdb/components/ConfigEditor.tsx
Comment (ryantxu) [inline public/app/plugins/datasource/influxdb/components/ConfigEditor.tsx:89]: How is this actually used?  I expect it would need to be passed to https://github.com/influxdata/influxdb-client-go

**Review by papagian — APPROVED** [review]
I've tested and AFAICT works great! I have tested without TLS and by selecting TLS Client Auth without filling the necessary fields and it complains as expected (`tls: failed to find any PEM data in certificate input`).
 It would also be nice to include a flux data source together with the other [gdev provisioned datasources](https://github.com/grafana/grafana/blob/master/devenv/datasources.yaml) by adding:
```
  - name: gdev-flux 
    type: influxdb
    access: proxy
    url: http://localhost:8086
    secureJsonData:
      token: mytoken
    jsonData:
      version: Flux
      organization: myorg
      defaultBucket: mybucket
```

**Comment by ryantxu** [comment]
I don't get how this does anything?  are we using the frontend proxy for anything with flux?

File: public/app/plugins/datasource/influxdb/components/ConfigEditor.tsx
Comment (aocenas) [reply public/app/plugins/datasource/influxdb/components/ConfigEditor.tsx:89]: We create the http client the same way for flux and then pass it to the client lib here https://github.com/grafana/grafana/blob/a2a6b9448ff0516a50738ac53fea28a3365d016f/pkg/tsdb/influxdb/flux/flux.go#L84. So I assume all the standard HTTP settings for the transport are respected.

**Pull request merged**
